<a href="https://colab.research.google.com/github/monicafdl/Prova2/blob/master/Prova_AV2_Gesta%CC%83o_do_Conhecimento.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

DISCIPLINA: Gestão do Conhecimento	 PROFESSOR(A): Alex Salgado
PERÍODO: 7o. TURNO: noite AVALIAÇÃO:

ALUNO(A): Mônica e Fernanda

GRAU: VISTO DO PROFESSOR:


# Questão 1 - valor (1,0)


*** Crie um repositório no Github e adicione este arquivo no mesmo


1.1 - Você deve escolher uma base de dados aberta que seja possível treinar um modelo de aprendizagem de máquina Supervisionado de Classificação (assim como o exemplo da Iris e Cancer). A partir desta base de dados, você vai treinar o modelo, fazer previsões e calcular a acurácia do mesmo.
Obs.: Não utilizar nenhuma base de dados usada em aula,  nos trabalhos ou na biblioteca do sklearn.


[link text](https://)

1.2 - Após tratada, criar uma pasta no seu GoogleDrive e disponibilizar o link do arquivo de sua base (csv, xlsx, etc) para ser importado no seu código.

https://drive.google.com/drive/folders/1jdI9XsP2Bg75CyA0K6NX4HPfnAc1m_Ey?usp=sharing

## Criar um código usando o Jupyter Notebook e responder (através de código) às seguintes questões:


# Questão 2 - valor (1,0)

2.1 - Importar os modulos python para machine learn e carregar o arquivo 

** Sugestão se for do tipo xlsx, usar o read_excel do pandas
import pandas as pd
dt = pd.read_excel("meuarquivo.xlsx")


In [ ]:
import pandas as pd

dt = pd.read_csv('dataR2.csv')

2.2 - Exibir os primeiros registros desta tabela.


In [ ]:
dt.head()

Age        BMI  Glucose  ...  Resistin    MCP.1  Classification
0   48  23.500000       70  ...   7.99585  417.114               1
1   83  20.690495       92  ...   4.06405  468.786               1
2   82  23.124670       91  ...   9.27715  554.697               1
3   68  21.367521       77  ...  12.76600  928.220               1
4   86  21.111111       92  ...  10.57635  773.920               1

[5 rows x 10 columns]

2.3 - De que se trata esse banco de dados? e que tipo de previsão pode ser feito com ele?


Existem 10 preditores, todos quantitativos e uma variável dependente binária, indicando a presença ou ausência de câncer de mama.
Os preditores são dados e parâmetros antropométricos que podem ser coletados em análises de sangue de rotina.
Modelos de previsão baseados nesses preditores, se precisos, podem ser potencialmente usados ​​como um biomarcador de câncer de mama.

In [ ]:
dt.info

<bound method DataFrame.info of      Age        BMI  Glucose  ...  Resistin    MCP.1  Classification
0     48  23.500000       70  ...   7.99585  417.114               1
1     83  20.690495       92  ...   4.06405  468.786               1
2     82  23.124670       91  ...   9.27715  554.697               1
3     68  21.367521       77  ...  12.76600  928.220               1
4     86  21.111111       92  ...  10.57635  773.920               1
..   ...        ...      ...  ...       ...      ...             ...
111   45  26.850000       92  ...  10.96000  268.230               2
112   62  26.840000      100  ...   7.32000  330.160               2
113   65  32.050000       97  ...  10.33000  314.050               2
114   72  25.590000       82  ...   3.27000  392.460               2
115   86  27.180000      138  ...   4.35000   90.090               2

[116 rows x 10 columns]>

# Questão 3 - valor (0,5)


Utilizando as terminologias de Machine Learning(features e observações):



3.1 - Quantas "features" têm nessa base de dados? R: 9 features.


Idade (anos)
IMC (kg / m2)
Glicose (mg / dL)
Insulina (µU / mL)
HOMA
Leptina (ng / mL)
Adiponectina (µg / mL)
Resistina (ng / mL) Resistina (ng / mL)
MCP-1 (pg / dL)

3.2 - Quantas observações têm nessa base de dados? R: 116 obervações.


In [ ]:
dt.shape

(116, 10)

# Questão 4 - valor (1,0)


4.1 - Faça uma previsão, usando o algoritmo de LogisticRegression.


In [ ]:
x = dt.iloc[:, 0:9]
y = dt.iloc[:, 9]

In [ ]:
from sklearn.linear_model import LogisticRegression
logreg = LogisticRegression(max_iter=5000)
logreg.fit(x, y)
logreg.predict(x)

array([1, 1, 1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
       1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 1, 2, 2, 2, 2, 2, 2, 2,
       2, 2, 2, 2, 2, 2, 2, 1, 2, 2, 2, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2,
       2, 1, 2, 1, 1, 2])

# Questão 5 - valor (1,5)


Usando o método de avaliação de acurácia (Treinar e testar na base de dados inteira/Train test entire model), Calcular a acurácia de cada um dos 3 métodos abaixo.

4.1 - Acurácia usando o algoritmo de KNN (com 1 vizinho, k=1)


In [ ]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics
knn = KNeighborsClassifier(n_neighbors=1)
knn.fit(x, y)
y_pred = knn.predict(x)
print(metrics.accuracy_score(y, y_pred))

1.0


4.2 - Acurácia usando o algoritmo de KNN (com 5 vizinho, k=5)


In [ ]:
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(x, y)
y_pred = knn.predict(x)
print(metrics.accuracy_score(y, y_pred))

0.7068965517241379


4.3 - Acurácia usando o algoritmo de LogisticRegression


In [ ]:
logreg = LogisticRegression(max_iter=5000)
logreg.fit(x,y)
y_pred= logreg.predict(x)
print(metrics.accuracy_score(y, y_pred))

0.8017241379310345


4.4 - De acordo com seus resultados anteriores, qual dos 3 métodos é mais eficiente?


In [ ]:
A acurácia usando o algoritmo de KNN com 1 vizinho foi o mais eficiente

In [ ]:
*** Envie o link do seu Github com essa resposta no Classroom
